src/PluginClientFactory.php

```diff
@@ -33,9 +32,9 @@ public static function setFactory(callable $factory)
     }
 
     /**
-     * @param HttpClient|HttpAsyncClient $client
-     * @param Plugin[]                   $plugins
-     * @param array                      $options {
+     * @param ClientInterface|HttpAsyncClient $client
```

Comment:
Missing import?

Reply:
https://github.com/GrahamCampbell/Analyzer would have detected that. It can be setup simply as a phpunit test: https://github.com/GrahamCampbell/Laravel-GitHub/blob/master/tests/AnalysisTest.php.

Reply:
nice! 
phpstan would also detect this i think, and some other semantic mistakes. on a high enough strictness even things like type mixups as in https://github.com/php-http/discovery/pull/112

Reply:
the import is fixed. if somebody wants to add the analyzer (or phpstan) lets do that in a separate pull request.